### PR TITLE
feat(modal): Ingested changes from rc-dialog to disable modal close button

### PR DESCRIPTION
**Pull request by Ke1sy**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

In this PR to rc-dialog https://github.com/react-component/dialog/pull/416 I added the a new prop for modal closable `closable={{ disabled: true }}` that allows to disable the close button.

This prop is useful when the content inside dialog includes long-running API call that is made upon clicking the “ok” button and need to prevent the user from closing the modal while the API call is still pending.
Currently there is no way to disable button while the API is still in flight

### 💡 Background and Solution

In this PR I made changed to able to use this new functionality inside antd Modal.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Added modal prop to disable close button |
| 🇨🇳 Chinese |  添加模态属性以禁用关闭按钮  |

ref: UIEN-6341

### CODE REVIEW

File: components/_util/hooks/useClosable.tsx [hunk 1]
```diff
@@ -3,7 +3,10 @@ import React from 'react';
 import CloseOutlined from '@ant-design/icons/CloseOutlined';
 import pickAttrs from 'rc-util/lib/pickAttrs';
 
-export type BaseClosableType = { closeIcon?: React.ReactNode } & React.AriaAttributes;
+export type BaseClosableType = {
+  closeIcon?: React.ReactNode;
+  disabled?: boolean;
+} & React.AriaAttributes;
 export type ClosableType = boolean | BaseClosableType;
```
Comment (afc163) [inline components/_util/hooks/useClosable.tsx:10]: Could we import this type from rc-dialog?

**Comment by Ke1sy** [comment]
@afc163 Ready, added type from rc-dialog, the "Preview Build" is failing not because of my changes (tested on separate branch.)

**Comment by Ke1sy** [comment]
@afc163 please, review. Thank you!

File: components/_util/hooks/useClosable.tsx [hunk 2]
```diff
@@ -104,10 +104,12 @@ export default function useClosable(
      */
     closeIconRender?: (closeIcon: ReactNode) => ReactNode;
   } = EmptyFallbackCloseCollection,
-): [closable: boolean, closeIcon: React.ReactNode | null] {
+): [closable: boolean, closeIcon: React.ReactNode | null, closeBtnIsDisabled: boolean] {
```
Comment (afc163) [inline components/_util/hooks/useClosable.tsx:107]:
```suggestion
): [closable: boolean, closeIcon: React.ReactNode, closeBtnIsDisabled: boolean] {
```

Comment (Ke1sy) [reply components/_util/hooks/useClosable.tsx]: @afc163 fixed

**Review by afc163 — APPROVED** [review]

**Pull request merged**